tools/lib/getDefaultWriteCallback.js

```diff
@@ -1,18 +1,13 @@
 'use strict';
-var fsExtra = require('fs-extra');
-var path = require('path');
-var Promise = require('bluebird');
-
-var fsExtraOutputFile = Promise.promisify(fsExtra.outputFile);
+var writeTile = require('./writeTile');
 
 module.exports = getDefaultWriteCallback;
 
 /**
  * @private
  */
-function getDefaultWriteCallback(outputDirectory) {
-    return function(file, data) {
-        var outputFile = path.join(outputDirectory, file);
-        return fsExtraOutputFile(outputFile, data);
+function getDefaultWriteCallback() {
+    return function(file, data, options) {
+        return writeTile(file, data, options);
```

Comment:
I think it's best to just keep this really simple like it was before where it just gets data and write to a file. Gzipping or anything else should be handled by the calling code. Maybe this makes `writeTile` not needed anymore.